### PR TITLE
Handle paginated check_runs API in api/taskcluster/webhook.go

### PR DESCRIPTION
**Pull request by stephenmcgruer**

### CODE REVIEW

**Comment by stephenmcgruer** [comment]
WDYT? I could definitely see trying to pull some logic out of ListCheckRuns, to make the pagination testable. Happy to put the work in to do that, if it feels necessary.

File: api/taskcluster/webhook.go [hunk 1]
```diff
@@ -431,13 +431,37 @@ func (api apiImpl) GetTaskGroupInfo(rootURL string, groupID string) (*TaskGroupI
...
+	hitEnd := false
+	for i := 0; i < 5; i++ {
+		result, response, err := api.ghClient.Checks.ListCheckRunsCheckSuite(api.ctx, owner, repo, checkSuiteID, &options)
+		if err != nil {
+			return runs, err
+		}
+
+		runs = append(runs, result.CheckRuns...)
+
+		// Setup for the next call.
```
Comment (Hexcles) [inline api/taskcluster/webhook.go:453]: Move this comment after the if block.

File: api/taskcluster/webhook.go [hunk 2]
```diff
@@ -431,13 +431,37 @@ func (api apiImpl) GetTaskGroupInfo(rootURL string, groupID string) (*TaskGroupI
...
+		if response.NextPage >= response.LastPage {
+			hitEnd = true
```
Comment (Hexcles) [inline api/taskcluster/webhook.go:455]: You could simply return runs here and you won't need `hitEnd`.

File: api/taskcluster/webhook.go [hunk 3]
```diff
@@ -431,13 +431,37 @@ func (api apiImpl) GetTaskGroupInfo(rootURL string, groupID string) (*TaskGroupI
 	return &group, nil
 }
 
-func (api apiImpl) ListCheckRuns(owner string, repo string, checkSuiteID int64) (*github.ListCheckRunsResults, *github.Response, error) {
+func (api apiImpl) ListCheckRuns(owner string, repo string, checkSuiteID int64) ([]*github.CheckRun, error) {
+	var runs []*github.CheckRun
```
Comment (Hexcles) [inline api/taskcluster/webhook.go:435]: Nit: maybe add a comment here saying 100 is the maximum allowed.

File: api/taskcluster/webhook.go (hunk 2 above)
Comment (stephenmcgruer) [reply api/taskcluster/webhook.go]: Lol, duh, thanks 🤣

**Pull request merged**